packages/nc-gui/components/smartsheet/Grid.vue

```diff
@@ -125,10 +128,65 @@ openNewRecordFormHook?.on(async () => {
 })
 
 const selectCell = (row: number, col: number) => {
+  clearRangeRows()
   selected.row = row
   selected.col = col
 }
 
+const selectBlock = (row: number, col: number) => {
+  if (selected.col === null || selected.row === null) {
+    if(isSelectedBlock){
+      selectedRows.endcol = col;
+      selectedRows.endrow = row;
+    }
+  }else if(selected.col !== col || selected.row !== row){
+    if(isSelectedBlock){
+      selected.col = null;
+      selected.row = null;
+      selectedRows.endcol = col;
+      selectedRows.endrow = row;
+    }
+  }
```

Comment:
Why check `if (isSelectedBlock)` in each case instead of checking it before all cases?

```ts
if (isSelectedBlock) {
   if (selected.col === null || selected.row === null) { ... }
   else if ( ... ) { ... }
}